storage/src/adb/current.rs

```diff
@@ -540,6 +1128,43 @@ pub mod test {
         });
     }
 
+    /// Repeatedly update the same key to a new value and ensure we can prove its current value
+    /// after each update.
+    #[test_traced("WARN")]
+    pub fn test_current_db_proving_repeated_updates() {
+        let executor = deterministic::Runner::default();
+        executor.start(|context| async move {
+            let mut hasher = Sha256::new();
+            let partition = "build_small";
+            let mut db = open_db(context.clone(), partition).await;
+
+            // Add one key.
+            let k = Sha256::fill(0x00);
+            for i in 1u8..=255 {
+                let v = Sha256::fill(i);
+                db.update(k, v).await.unwrap();
+                assert_eq!(db.get(&k).await.unwrap().unwrap(), v);
+                db.commit().await.unwrap();
+                let root = db.root(&mut hasher).await.unwrap();
+
+                // Create a proof for the current value of k.
+                let (proof, info) = db.key_value_proof(&mut hasher, k).await.unwrap();
+                assert_eq!(info.value, v);
+                assert!(
```

Comment:
nit: might be nice to add an assertion that proving at the last updated value (or all previous) fails at each step?

I don't think this will improve coverage but seems like a nice feature of this test.